### PR TITLE
[lib] Harmonize index entries for 'make_error_code' and 'make_error_c…

### LINKED ISSUE #621: Index subentries for make_error_code and make_error_condition
```
$ grep indexlibrary.*make_error_code *.tex
diagnostics.tex:\indexlibrary{\idxcode{make_error_code}}%
diagnostics.tex:\indexlibrary{\idxcode{make_error_code}}
iostreams.tex:\indexlibrary{\idxcode{make_error_code}}%
threads.tex:\indexlibrary{\idxcode{make_error_code}!\idxcode{future_errc}}%
threads.tex:\indexlibrary{\idxcode{future_errc}!\idxcode{make_error_code}}%
```

This means the index looks like:

```
make_error_code, 500, 506, 1033
    future_errc, 1197
```

And similarly for make_error_condition:

```
make_error_condition, 500, 508, 1033
    future_errc, 1197
```

We should either use subentries for the argument type on all overloads or none.

### PR DESCRIPTION
**Pull request by jensmaurer**
…ondition'.

Fixes #621.

### CODE REVIEW

File: source/iostreams.tex [hunk 1]
```diff
@@ -626,6 +626,7 @@
 \synopsis{Header \tcode{<ios>} synopsis}
```
Comment (tkoeppe) [inline source/iostreams.tex]: Eew. This synopsis should *replace* the rSec2 heading just above, but the stable name is a bit silly then.

Comment (tkoeppe) [inline source/iostreams.tex]: I can merge this as is, but perhaps something worth cleaning up. Let me know how you want to proceed.

Comment (jensmaurer) [inline source/iostreams.tex]: @tkoeppe: I sympathize, but that issue is not the subject of this pull request. Feel free to vent your anger in #566.

Comment (tkoeppe) [inline source/iostreams.tex]: I mean, you seem to enjoy numbering synopses... :-)

**Pull request merged**